### ISSUE TITLE
Unable to add a existing Cluster with the Admin role in the Rancher V2.5.1

### ISSUE THREAD

**Issue by allaboutopensource** [rancher/rancher#29451]
What kind of request is this (question/bug/enhancement/feature request): bug

Steps to reproduce (least amount of steps as possible):
Installed the latest version of Rancher server 2.5.1
Trying to adding an existing cluster using the Admin role getting an error: 
Failed while: Wait for Condition: InitialRolesPopulated: True

Tried creating new user with the restricted Admin role but getting an error: 
Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: context deadline exceeded

kubernetes version : v1.18.8
VM version : CentOS Linux 7 (Core) and kernel version : 3.10.0-1062.el7.x86_64
Result:
Unable to create Cluster using the Admin role
Unable to create user with Restricted ADMIN role 

Other details that may be helpful:
Though we are getting an error while creating the user, i can see the user being created but unable to login with the user credentials
Though we are getting an error while adding the cluster, we see the cluster is created but in Pending status. 

Environment information

Rancher version (rancher/rancher/rancher/server image tag or shown bottom left in the UI): 2.5.1
Installation option (single install/HA): HA
Cluster information

Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Its running on vmware
Machine type (cloud/VM/metal) and specifications (CPU/memory): VM with 8gb RAM and 8 core Vcpus
Kubernetes version (use kubectl version): v1.18.8
(paste the output here)
Docker version (use docker version): v18.9.2

**Comment by dramich** [comment]
@allaboutopensource There is an issue with your webhook responding within 10 seconds. So things to check, did it get deployed? Are you running a proxy?

**Comment by allaboutopensource** [comment]
i do see that the webhook container is running without any issues. 

![image](https://user-images.githubusercontent.com/5890851/96332383-4a66aa80-1081-11eb-8ca5-b231bff2595c.png)

**Comment by zhcsyncer** [comment]
reproduce after upgrade from 2.5.1 to v2.5.2

Environment information

Rancher version (rancher/rancher/rancher/server image tag or shown bottom left in the UI): 2.5.2
Installation option (single install/HA): HA
Cluster information

Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted
Kubernetes version (use kubectl version): 
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.3", GitCommit:"06ad960bfd03b39c8310aaf92d1e7c12ce618213", GitTreeState:"clean", BuildDate:"2020-02-13T18:08:14Z", GoVersion:"go1.13.8", Compiler:"gc", Platform:"darwin/amd64"}

Server Version: version.Info{Major:"1", Minor:"16+", GitVersion:"v1.16.9-aliyun.1", GitCommit:"4f7ea78", GitTreeState:"", BuildDate:"2020-05-08T07:29:59Z", GoVersion:"go1.13.9", Compiler:"gc", Platform:"linux/amd64"}```

**Comment by andreobrown** [comment]
I'm seeing the same issue: I am unable to Import and Amazon EKS cluster.

Rancher Version: 2.5.2

![Screen Shot 2020-11-21 at 00 46 27](https://user-images.githubusercontent.com/1885207/99868748-49023380-2b93-11eb-8f1d-08eb6aea87b5.png)

`kubectl version                                                                                                                                                                                                                                                                           

Server Version: version.Info{Major:"1", Minor:"18+", GitVersion:"v1.18.9-eks-d1db3c", GitCommit:"d1db3c46e55f95d6a7d3e5578689371318f95ff9", GitTreeState:"clean", BuildDate:"2020-10-20T22:18:07Z", GoVersion:"go1.13.15", Compiler:"gc", Platform:"linux/amd64"}`

In the cluster section of the dashboard, the cluster seems to be added (with an authorisation error that I'm now troubleshooting)

![Screen Shot 2020-11-21 at 00 52 30](https://user-images.githubusercontent.com/1885207/99868814-eb221b80-2b93-11eb-9d2d-72ff3801d10a.png)

**Comment by kherrmann3** [comment]
I'm seeing the same error as the original post, but not seeing anything in the rancher-webhook or rancher pod logs. I am getting a web socket error when I login to Rancher, though. 

Here is the error on the rancher server logs: Internal error occurred: failed calling webhook "rancherauth.cattle.io": Post https://rancher-webhook.cattle-system.svc:443/v1/webhook/validation?timeout=10s: context deadline exceeded

In my case, I'm running Rancher 2.5.5 on GKE, behind nginx ingress controller, which is terminating the TLS connection. There is a TCP LB in front of rancher, but shouldn't be interfering with the traffic. 

From what I can tell, the rancher pods should be connecting to the rancher-webhook service behind the ingress controller, is that correct? If so, the ingress controller shouldn't be causing any issues. That said, I don't see either of the deployments mounting any ca certs. In fact, the rancher deploy has the --no-cacerts argument. I do see that a cattle-webhook-ca and cattle-webhook-tls secret were created, but don't appear to be referenced in any manifests. Do I need to ensure those are mounted? I did exec into the rancher pod and curled "https://rancher-webhook.cattle-system.svc:443", and it gave an SSL error "unable to get local issuer certificate". So, this is where I'm assuming the issue is likely at.

I'll see about manually mounting the rancher-webhook-ca in the rancher pod, but it also doesn't appear that the rancher-webhook itself is mounting its TLS secret, so not sure how this is all supposed to gel together. Should the webhook have its own TLS cert mounted?

Thank you.

**Comment by bluemalkin** [comment]
@kherrmann3 I've ran into exactly the same issues described here. I couldn't import clusters with WSS timeout, couldn't add resources in rancher-server UI  (timeout as well). I spent days thinking it was an issue with the Internal TCP load balancer (created from nginx-ingress). 

You are 100% correct it was a missing FW rule between the master `/28` range abd the webbook pods port `9443`.

As soon as I added that rule bang everything worked.

Hope this saves the pain for others too.
